Model/RatepayInvoicePaymentMethod.php

```diff
@@ -60,6 +62,25 @@ public function getTransaction()
         return new RatepayInvoiceTransaction();
     }
 
+    /**
+     * @inheritdoc
+     * @param Transaction $oTransaction
+     * @param Order       $oOrder
+     *
+     * @since 1.2.0
+     */
+    public function addMandatoryTransactionData(&$oTransaction, $oOrder)
+    {
+        $oSession = Registry::getSession();
+        $oBasket = $oSession->getBasket();
+        $oWdBasket = $oBasket->createTransactionBasket();
+
+        $oTransaction->setBasket($oWdBasket);
+        $oTransaction->setAccountHolder($oOrder->getAccountHolder());
+        $oTransaction->setShipping($oOrder->getShippingAccountHolder());
+        $oTransaction->setOrderNumber($oOrder->oxorder__oxid->value);
```

Comment:
Is the order number not added so or so for each payment? And if so there is no need to do it again in this method?

Reply:
It is actually bound to whether or not additional data should be sent. I changed it accordingly.